util/src/main/java/io/kubernetes/client/util/Threads.java

```diff
@@ -20,9 +20,14 @@ public class Threads {
 
   public static ThreadFactory threadFactory(String format) {
     final ThreadFactory defaultFactory = Executors.defaultThreadFactory();
+    // Users can inject their own thread 
+    public static void setDefaultThreadFactory(ThreadFactory factory) {
+      defaultFactory = factory;
```

Comment:
this doesn't work correctly because you use `factory` below.

If you remove the change on line 27 (`final ThreadFactory factory ...`) and also the change on line 30 (`Thread thread = factory.newThread(r);`) then I think this code will work correctly.

However, please add a unit test to confirm the behavior is working as intended.